Sources/Sentry/SentryScope.m

```diff
@@ -461,14 +461,15 @@ - (SentryEvent *__nullable)applyToEvent:(SentryEvent *)event
         @synchronized(_spanLock) {
             span = self.span;
         }
-        if (![event.type isEqualToString:SentryEnvelopeItemTypeTransaction] &&
-            [span isKindOfClass:[SentryTracer class]]) {
-            event.transaction = [(SentryTracer *)span name];
-        }
-        if (newContext == nil) {
-            newContext = [NSMutableDictionary new];
+
+        // Span could be nil as we do the first check outside the synchronize
+        if (span != nil) {
+            if (![event.type isEqualToString:SentryEnvelopeItemTypeTransaction] &&
+                [span isKindOfClass:[SentryTracer class]]) {
+                event.transaction = [(SentryTracer *)span name];
+            }
+            newContext[@"trace"] = [span serialize];
```

Comment:
The first nil check, sets context = event.context, which is a nullable property, I guess it may continue nil after that.